### PR TITLE
Add tailwind support and refactor dashboard to functional components

### PR DESCRIPTION
**Pull request by nezdemkovski**
I am currently migrating the risk plugin to version 2.4 and I have rewritten it using pure react + tailwind. As a result, I needed to modify the template slightly to ensure that all styles work well. Consequently I decided to incorporate these changes into the main template as well.

- I have added basic tailwind support and isolated it from the main application styles by applying it only to the `.addon` container. There appear to be no conflicts between bootstrap and tailwind. While it's optional to use tailwind doing so can simplify the styling process during the migration or creation of new plugins.
- I have refactored some dashboard components to use modern functional components instead of the older class-based approach.
- Bumped the deps to latest

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -22,35 +22,69 @@ module.exports = {
...
+                                            if (selector.startsWith("html") || selector.startsWith("body")) {
+                                                return selector;
+                                            }
+                                            return prefixedSelector;
+                                        },
+                                    }),
+                                    require("tailwindcss"),
+                                    require("autoprefixer"),
+                                ],
+                            },
```
Comment (sirenevenkii) [inline webpack.config.js:71]: can we utilise `postcss.config.js` here?

**Review by sirenevenkii — COMMENTED** [review]
tbh, I'm a little bit hesitant to add such a big library, see https://bundlephobia.com/package/tailwindcss@3.4.1. most of it is actually dev dependencies, but even if we just look at the "self" part, it's still 200kb, which is quite hard to justify for relatively small use case.
in ideal world we should provide users with a components that they would be able to use within the plugin. but even considering we're not there yet, I don't think that adding relatively big css lib is a good idea even in the short term as an interim solution.

**Comment by francoischasseur** [comment]
Can we have tailwind as a peerDependencies ( like we do for react ?) So we include once and share it with the whole plugin ecosystem ?

**Comment by mike-matrixreq** [comment]
I like that we're collecting a variety of approaches to writing these plugins...maybe we do a retrospective across them soon and identify the best approaches, targeting exactly those for boilerplate inclusion.

I see a task to document and make a video about the boilerplate project and how to use it in our future.

File: webpack.config.js [hunk 2]
```diff
@@ -22,35 +22,69 @@ module.exports = {
...
+                        },
+                    },
+                    {
+                        loader: "postcss-loader",
+                        options: {
+                            postcssOptions: {
+                                plugins: [
+                                    require("postcss-import"),
+                                    require("postcss-prefix-selector")({
+                                        // that's for tailwind not to mess with bootstrap styles
```
Comment (mike-matrixreq) [inline webpack.config.js:59]: +1 on the careful thought integrating with the legacy main app...

**Comment by nezdemkovski** [comment]
@sirenevenkii @francoischasseur That's not really how the tailwind works, we cannot put it as a peerDep. Tailwind is basically a postcss plugin with [JIT](https://tailwindcss.com/blog/just-in-time-the-next-generation-of-tailwind-css) engine enabled, that means it generates styles based on your use. It's works differently than bootstrap where you need to have the whole css attached to your web. So if you check the output you'll find out that it generated the css with classes I use in the project only.

Theoretically, it's possible to put the part of the tailwind base styles (that's usually just reset styles) into main app so we don't share the same css over and over again.

**Comment by nezdemkovski** [comment]
@sirenevenkii @francoischasseur I just checked the size of the bundle with tailwind styles and without them for comparison. 
First of all, our current webpack config doesn't optimize bundles for production mode and we have reeeeally big bundles for each plugin (I'll send a PR with webpack fixes later).

Now to our numbers:
- optimization mode off: 362KB
- optimization mode off + tailwind: 395KB (24KB is basic tw styles here, pure tw styles are 7KB)
- optimization mode on: 105KB
- optimization mode on + tailwind: 136KB (24KB is basic tw styles here, pure tw styles are 7KB)

If we move tailwind basic styles to separate file that will save us 24KB from each bundle.

File: webpack.config.js (hunk 1 above)
Comment (nezdemkovski) [reply webpack.config.js:71]: yeah, we can and we should, but magically it works only when I duplicate that thing 😅 Removing postcss config or changing `postcss-prefix-selector` in webpack just break it. I'll investigate later

**Review by sirenevenkii — APPROVED** [review]
thanks for explanation, never worked with tailwind before and wasn't aware of the JIT. 
LGTM

**Pull request closed without merging**